entrypoint.go

```diff
@@ -16,26 +16,37 @@ package main
 
 import (
 	"context"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
 	"os/signal"
 	"path/filepath"
 	"syscall"
+	"time"
 
 	"github.com/GoogleCloudPlatform/run-gmp-sidecar/confgenerator"
+	"gopkg.in/yaml.v2"
 )
 
 // Create channel to listen for signals.
 var signalChan chan (os.Signal) = make(chan os.Signal, 1)
 var userConfigFile = "/etc/rungmp/config.yaml"
 var otelConfigFile = "/run/rungmp/otel.yaml"
+var configRefreshInterval = 10 * time.Second
+var selfMetricsPort = 0
 
-func main() {
-	// SIGINT handles Ctrl+C locally.
-	// SIGTERM handles Cloud Run termination signal.
-	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
-	ctx := context.Background()
+// Generate OTel config from RunMonitoring config. Returns FileInfo of the user
+// config, whether its the default config, and an error if generation of OTel
+// configs failed.
+func generateOtelConfig(ctx context.Context, userConfigFile string) (os.FileInfo, bool, error) {
+	info, err := os.Stat(userConfigFile)
+	if os.IsNotExist(err) {
+		return nil, true, nil
+	}
+	if err != nil {
+		return nil, false, err
+	}
```

Comment:
This part of the code is repeated later (Line 113) and could be separated out to its own function, so that  `generateOtelConfig()` can focus on generating and writing the config. Here is what I have in mind: 

```
func getConfigFileInfo(userConfigFile string) (os.FileInfo, bool, error) {
	info, err := os.Stat(userConfigFile)
	if os.IsNotExist(err) {
		return nil, true, nil
	}
	if err != nil {
		return nil, false, err
	}
	return info, false, nil
}
...
func generateOtelConfig(ctx context.Context, userConfigFile string) error {
...
}
...

```

Reply:
Done